crates/rust-project-goals-cli/src/updates.rs

```diff
@@ -37,7 +36,101 @@ pub(crate) fn generate_updates(
         );
     }
 
-    let issues = list_issues_in_milestone(repository, milestone)?;
+    let issues = cached_issues;
+
+    // Load goal documents to extract theme information and optionally filter by team
+    let mut milestone_path = PathBuf::from("src");
+    milestone_path.push(milestone);
+    let goal_documents = goal::goals_in_dir(&milestone_path)?;
+
+    // Create a mapping from issue numbers to themes for flagship goals
+    let issue_themes: std::collections::HashMap<u64, String> = goal_documents
+        .iter()
+        .filter_map(|doc| {
+            if let (Some(flagship_theme), Some(tracking_issue)) = 
+                (doc.metadata.flagship(), doc.metadata.tracking_issue.as_ref()) {
+                Some((tracking_issue.number, flagship_theme.trim().to_string()))
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    // Create mappings for ownership information
+    let issue_point_of_contact: std::collections::HashMap<u64, String> = goal_documents
+        .iter()
+        .filter_map(|doc| {
+            doc.metadata.tracking_issue.as_ref().map(|issue| {
+                (issue.number, doc.point_of_contact_for_goal_list())
+            })
+        })
+        .collect();
+
+    let issue_team_champions: std::collections::HashMap<u64, String> = goal_documents
+        .iter()
+        .filter_map(|doc| {
+            doc.metadata.tracking_issue.as_ref().map(|issue| {
+                let teams: std::collections::BTreeSet<&rust_project_goals::team::TeamName> = doc
+                    .team_asks
+                    .iter()
+                    .flat_map(|ask| &ask.teams)
+                    .copied()
+                    .collect();
+
+                let team_champions: Vec<String> = teams
+                    .into_iter()
+                    .filter_map(|team| {
+                        doc.metadata.champions.get(team).map(|champion| {
+                            format!("{} ({})", team, champion.content)
+                        })
+                    })
+                    .collect();
+
+                let team_champions_str = if team_champions.is_empty() {
+                    "(none)".to_string()
+                } else {
+                    team_champions.join(", ")
+                };
+
+                (issue.number, team_champions_str)
+            })
+        })
+        .collect();
+
+    let issue_task_owners: std::collections::HashMap<u64, String> = goal_documents
+        .iter()
+        .filter_map(|doc| {
+            doc.metadata.tracking_issue.as_ref().map(|issue| {
+                let task_owners_str = if doc.task_owners.is_empty() {
+                    "(none)".to_string()
+                } else {
+                    doc.task_owners.iter().cloned().collect::<Vec<_>>().join(", ")
```

Comment:
The typesystem docs goal, and the evolving hierarchy goal, show an issue for task owners, here or in the goal document parsing (maybe due to the lack of "," separating some owners) or regexes -- but we can have duplicated task owners here. I'm not sure where we should be between structured parsing and flexibility in the text.

The RFL goal for compiler features data also looks incorrect here.